cli/purge.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+func MakePurgeCommand() *cobra.Command {
+	var force bool
+	var cmd = &cobra.Command{
+		Use:   "purge",
+		Short: "Delete all persisted DefraDB data",
+		Long: `Delete all persisted DefraDB data.
+WARNING this operation will delete all data and cannot be reversed.`,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := setContextRootDir(cmd); err != nil {
+				return err
+			}
+			return setContextConfig(cmd)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !force {
+				return ErrPurgeForceFlagRequired
+			}
+			cfg := mustGetContextConfig(cmd)
+			return os.RemoveAll(cfg.GetString("datastore.badger.path"))
```

Comment:
thought: This will only clear the badger directory, if the user is using another datastore this command will not work (unclear if will panic, or quietly do nothing), and the added tests do not automatically cover other dbs.  I cannot remember how hard-coded other CLI commands are to badger, this might be a drop in the ocean when it comes to adding support for other datastores in the CLI.

Reply:
When we add more datastores we can have them all use the same `datastore.path` as their root path and have sub directories for each datastore type. Then we just need to delete the root path to remove them all.